src/nvim/api/buffer.c

```diff
@@ -514,24 +514,26 @@ void nvim_buf_set_lines(uint64_t channel_id, Buffer buffer, Integer start, Integ
 
 /// Sets (replaces) a range in the buffer
 ///
-/// This is recommended over nvim_buf_set_lines when only modifying parts of a
-/// line, as extmarks will be preserved on non-modified parts of the touched
+/// This is recommended over |nvim_buf_set_lines()| when only modifying parts of
+/// a line, as extmarks will be preserved on non-modified parts of the touched
 /// lines.
 ///
-/// Indexing is zero-based and end-exclusive.
+/// Row indices are zero-based and end-inclusive. Column indices are zero-based,
+/// end-exclusive, and correspond to bytes, not character columns (see |col()|
+/// vs. |charcol()|).
 ///
-/// To insert text at a given index, set `start` and `end` ranges to the same
-/// index. To delete a range, set `replacement` to an array containing
-/// an empty string, or simply an empty array.
+/// To insert text at a given `(row, column)` location, use `start_row = end_row
+/// = row` and `start_col = end_col = col`. To delete the text in a range, use
+/// `replacement = {}`.
 ///
-/// Prefer nvim_buf_set_lines when adding or deleting entire lines only.
+/// Prefer |nvim_buf_set_lines()| if you are only adding or deleting entire lines.
 ///
 /// @param channel_id
 /// @param buffer           Buffer handle, or 0 for current buffer
 /// @param start_row        First line index
-/// @param start_col        First column
-/// @param end_row          Last line index
-/// @param end_col          Last column
+/// @param start_col        Starting byte offset of first line
```

Comment:
```suggestion
/// @param start_col        Starting byte offset on first line
```

Reply:
I'd rather write something like `Starting column (byte offset) on first line` . "Column" _is_ the standard term for "byte offset relative to the start of line", and if this is meant to be an entry point for new API users it should be emphasized properly. ("byte offset" by itself is rather used for the offset in the entire buffer, or between lines)

Reply:
Ok. I've done a more thorough triage of terminology and I agree with this. More to come but for now I have incorporated this suggestion.